### ISSUE TITLE
Duplicate Key in Dictionary Literal with Computed Var

### ISSUE THREAD

**Issue by aetherealtech** [realm/SwiftLint#5077]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `duplicated_key_in_dictionary_literal` warning is triggered when one static computed var is used repeatedly in a dictionary literal, despite the var being computed and therefore not necessarily the same on every read.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
warning: Duplicated Key in Dictionary Literal Violation: Dictionary literals with duplicated keys will crash at runtime (duplicated_key_in_dictionary_literal)
```

### Environment

* SwiftLint version: 0.52.3
* Installation method used: Homebrew
* Paste your configuration file: N/A (default config)

```swift
// This triggers a violation:
var currentValue = 0
var generateNewValue: Int {
    let current = currentValue
    currentValue += 1
    return current
}

let testDict = [
    generateNewValue: "",
    generateNewValue: "",
    generateNewValue: "",
]
```

**Comment by SimplyDanny** [comment]
SwiftLint's normal rules operate on syntax-level only. Therefore, they don't know if `generateNewValue` is a constant or a computed property. Even with this information, it's unclear if computing the property has any side-effects, especially when the property is defined in another type that might even be imported from another module.

That said, the rule accepts function calls assuming they always produce a different value. It has to make compromises and `UUID()` was a good reason to exclude function calls.

Long story short, while this is a bug I think we have to accept the false-positives caused by code similar to yours.
